src/Map.cpp

```diff
@@ -65,22 +65,22 @@ void cMap::Tick()
 
 
 
-void cMap::UpdateRadius(int a_PixelX, int a_PixelZ, unsigned int a_Radius)
+void cMap::UpdateRadius(unsigned int a_PixelX, unsigned int a_PixelZ, unsigned int a_Radius)
 {
-	int PixelRadius = a_Radius / GetPixelWidth();
+	unsigned int PixelRadius = a_Radius / GetPixelWidth();
 
-	unsigned int StartX = Clamp(a_PixelX - PixelRadius, 0, (int)m_Width);
-	unsigned int StartZ = Clamp(a_PixelZ - PixelRadius, 0, (int)m_Height);
+	unsigned int StartX = Clamp(a_PixelX - PixelRadius, static_cast<unsigned int>(0), m_Width);
+	unsigned int StartZ = Clamp(a_PixelZ - PixelRadius, static_cast<unsigned int>(0), m_Height);
```

Comment:
Exactly this is the reason why I wrote that these changes will require very thorough examination :)